week-3/mandatory/2-api/cyf-ecommerce-api/server.js

```diff
@@ -0,0 +1,214 @@
+const express = require('express')
+const app = express()
+app.use(express.json())
+
+const { Pool } = require('pg')
+
+const db = new Pool({
+    user: 'ORHAN',
+    host: 'localhost',
+    database: 'cyf_ecommerce',
+    password: '*****',
+    port: 5432
+})
+
+
+app.get("/products", (req, res) => {
+    const searchName = req.query.name
+    searchName && searchName.toLowerCase()
+
+    const attributes = [searchName]
+    const sql1 = "select p.product_name, pa.unit_price, s.supplier_name from product_availability pa join products p on pa.prod_id = p.id join suppliers s on pa.supp_id = s.id  where Lower(p.product_name) like '%'|| $1 ||'%'";
+    const sql2 = 'select p.product_name, pa.unit_price, s.supplier_name from product_availability pa join products p on pa.prod_id = p.id join suppliers s on pa.supp_id = s.id';
+
```

Comment:
while this will work with no issue
it can be harder to maintain and scale up
for better fix you can have arguments and where conditions alone  as variables that are added to main string on a conditions